### ISSUE TITLE
[Idea] Spamming prevention

### ISSUE THREAD

**Issue by Nambaseking01** [micahlt/modchat#8]
**Is your feature request related to a problem? Please describe.**
Obviously spamming - it's very annoying to see that people can spam very easily. I thought of a prevention for this.

**Describe the solution you'd like**
Maybe there could be a system that users can only post the same thing 5 times in a row and then it would prevent them from posting that specific thing for the next 10 seconds. If it's possible, perhaps it could also be transferred onto other rooms if the user moved to other rooms to start spamming. 

**Additional context**
The GUI would probably work similar to the one suggested in issue #6

**Comment by Semisol** [comment]
This system is broken, just add a random number and a macro. Cooldowns?

**Comment by micahlt** [comment]
@Semisol Yeah, just like RNTAS on Scratch.  Cooldowns would be the most effective method.

**Comment by Semisol** [comment]
> @Semisol Yeah, just like RNTAS on Scratch. Cooldowns would be the most effective method.

Actually, use a Discord-like approach.
It at first imposes a small, 200ms like delay.
And if you keep spamming, depending on message similarity, speed between messages and other factors, your messages are delayed more until a point you have to wait for other messages to finish sending to send more.

**Comment by contourcode** [comment]
Another discord style mechanism could be a flagging process where if you have made lots of messages in a short period of time, you get locked out of modchat until a moderator unflags you?

**Comment by Semisol** [comment]
> Another discord style mechanism could be a flagging process where if you have made lots of messages in a short period of time, you get locked out of modchat until a moderator unflags you?

lockouts must have sufficient proof behind them and im busy with other stuff so no v2 :/

**Comment by micahlt** [comment]
Wait, so you can't work on V2?

**Comment by Semisol** [comment]
I don't think I can too much. I am trying to do personal projects, also why mail?
